timelock-server/src/integTest/java/com/palantir/atlasdb/timelock/MultiNodePaxosTimeLockServerIntegrationTest.java

```diff
@@ -342,6 +348,56 @@ public void distinctClientsStillShareTheSameSequenceOfTimestamps() {
         assertThat(temporalSequence).isSorted();
     }
 
+    @Test
+    public void temporalOrderingIsPreservedForBatchedStartTransactionRequests() {
+        UUID requestor = UUID.randomUUID();
+        List<Long> allTimestamps = new ArrayList<>();
+
+        allTimestamps.addAll(getSortedBatchedStartTimestamps(requestor, 1));
+        allTimestamps.addAll(getSortedBatchedStartTimestamps(requestor, 4));
+        allTimestamps.addAll(getSortedBatchedStartTimestamps(requestor, 20));
+
+        assertThat(allTimestamps).isSorted();
+    }
+
+    @Test
+    public void temporalOrderingIsPreservedBetweenDifferentRequestorsForBatchedStartTransactionRequests() {
+        UUID requestor = UUID.randomUUID();
+        UUID requestor2 = UUID.randomUUID();
+        List<Long> allTimestamps = new ArrayList<>();
+
+        allTimestamps.addAll(getSortedBatchedStartTimestamps(requestor, 1));
+        allTimestamps.addAll(getSortedBatchedStartTimestamps(requestor2, 4));
+        allTimestamps.addAll(getSortedBatchedStartTimestamps(requestor, 20));
+        allTimestamps.addAll(getSortedBatchedStartTimestamps(requestor2, 15));
+
+        assertThat(allTimestamps).isSorted();
+    }
+
+    @Test
+    public void batchedTimestampsShouldBeSeparatedByModulus() {
+        UUID requestor = UUID.randomUUID();
+
+        List<Long> sortedTimestamps = getSortedBatchedStartTimestamps(requestor, 10);
+
+        Set<Long> differences = IntStream.range(0, sortedTimestamps.size() - 1)
+                .mapToObj(i -> sortedTimestamps.get(i + 1) - sortedTimestamps.get(i))
+                .collect(Collectors.toSet());
+
+        assertThat(differences).containsOnly((long) TransactionConstants.V2_TRANSACTION_NUM_PARTITIONS);
+    }
+
+    private List<Long> getSortedBatchedStartTimestamps(UUID requestorUuid, int numRequestedTimestamps) {
+        BatchedStartTransactionRequest request = BatchedStartTransactionRequest.createForRequestor(
+                requestorUuid,
+                numRequestedTimestamps);
+        BatchedStartTransactionResponse response = CLUSTER.timelockRpcClient().batchedStartTransaction(request);
+        return response.timestamps().stream()
+                .boxed()
+                .sorted() // we do not guarantee sorted order for same batch
```

Comment:
we can probably just kill these two lines; the wire format means we effectively guarantee sort order...